app/src/routes/routesMap.js

```diff
@@ -66,35 +71,53 @@ const redirectRoute = (path, createNewAction) => ({
 });
 
 export const onBeforeRouteChange = (dispatch, getState, { action }) => {
-  const authorized = isAuthorizedSelector(getState());
-  if (!authorized) {
-    return;
-  }
-
   const {
-    payload: { projectId: hashProject },
     type: nextPageType,
+    payload: { projectId: hashProject },
   } = action;
-  if (!hashProject) {
-    return;
-  }
 
-  const currentPageType = pageSelector(getState());
-  const activeProjectId = activeProjectSelector(getState());
+  const authorized = isAuthorizedSelector(getState());
+  let projectId = activeProjectSelector(getState());
+  const accountRole = userAccountRoleSelector(getState());
   const userInfo = userInfoSelector(getState());
 
   const userProjects = userInfo ? userInfo.assignedProjects : {};
-  const isAdminNewPageType = !!adminPageNames[nextPageType];
-  const isAdminCurrentPageType = !!adminPageNames[currentPageType];
 
-  if (
-    userProjects &&
-    Object.prototype.hasOwnProperty.call(userProjects, hashProject) &&
-    (hashProject !== activeProjectId || isAdminCurrentPageType) &&
-    !isAdminNewPageType
-  ) {
+  if (userProjects && hashProject in userProjects && projectId !== hashProject) {
```

Comment:
In case when user on admin projects page view project details (we use `projectId` in the url) system shouldn't change `activeProject`.